internal/impl/generate.go

```diff
@@ -61,7 +61,16 @@ func (d *Devbox) generateShellFiles() error {
 	}
 
 	for _, pkg := range d.packages() {
-		if err := d.pluginManager.CreateFilesAndShowReadme(d.writer, pkg, d.projectDir); err != nil {
+		if err := d.pluginManager.Create(d.writer, pkg, d.projectDir); err != nil {
+			return err
+		}
+	}
+
+	for _, includes := range d.cfg.Include {
+		if err := d.lockfile.Add(includes); err != nil {
+			return err
+		}
+		if err := d.pluginManager.Include(d.writer, includes, d.projectDir); err != nil {
```

Comment:
It'll be good for some future PR  to make the plugin manager life-cycle more clear. 

Suggestion:
- `pluginManager.New(lockfile)` function that does `applyOptions`
- `pluginManager.Generate(packages, includes)`
   - this can do `lockfile.Add(includes)` internally

Reply:
`NewManager` already applies options. The reason we call it separately is because of a circular dependency (see `impl.Open` function)

Agree with unifying `Create` and `Includes`

Not sure about calling `lockfile.Add(includes)` internally. That feels like logic that is unrelated to plugins. Insterad, plugins should be saved to lock file and then we can pass in resolved packages directly to plugin manager, which will mean we no longer need the lockfile in there.

Reply:
> NewManager already applies options. The reason we call it separately is because of a circular dependency (see impl.Open function)

Oh I see. I left a comment above about improving that so its easier to reason about.

> Not sure about calling lockfile.Add(includes) internally. That feels like logic that is unrelated to plugins. Insterad, plugins should be saved to lock file and then we can pass in resolved packages directly to plugin manager, which will mean we no longer need the lockfile in there.

Yup, you're right. Lets not call lockfile.Add from within pluginManager. 👍🏾